solid/src/main/java/com/inrupt/client/solid/SolidClient.java

```diff
@@ -134,8 +138,19 @@ public <T extends Resource> CompletionStage<T> read(final URI identifier, final
         return client.send(request, Response.BodyHandlers.ofByteArray())
             .thenApply(response -> {
                 if (response.statusCode() >= ERROR_STATUS) {
-                    throw SolidClientException.handle("Unable to read resource at " + request.uri(), request.uri(),
-                            response.statusCode(), response.headers(), new String(response.body()));
+                    final ProblemDetails pd = ProblemDetails.fromErrorResponse(
+                        response.statusCode(),
+                        response.headers(),
+                        response.body(),
+                        this.jsonService
+                    );
+                    throw SolidClientException.handle(
+                        "Unable to read resource at " + request.uri(),
+                        pd,
+                        response.uri(),
+                        response.headers(),
+                        new String(response.body())
+                    );
```

Comment:
There are three of these blocks here. They are very similar.

Could they be extracted?

Reply:
This is now handled by the exception mapper.